src/Build/BackEnd/Shared/TargetResult.cs

```diff
@@ -44,17 +44,30 @@ public class TargetResult : ITargetResult, ITranslatable
         /// </summary>
         private CacheInfo _cacheInfo;
 
+        /// <summary>
+        /// The (possibly null) <see cref="BuildEventContext"/> from the original target build
+        /// </summary>
+        private BuildEventContext _originalBuildEventContext;
```

Comment:
Would it make sense to stick the SkipReason in TargetResult as well? Feels like the API would be more complete if a user can ask a TargetResult if it represents a skipped target or not. Or maybe have an OriginalTargetResult instead of the OriginalBuildEventContext, and then put a BuildEventContext in all of them?

Reply:
Can a TargetResult represent a skipped target? I thought it always represents the original target?

Reply:
A BuildEventContext is a well-known "currency" type that can float around everywhere and is understood by node packet translator, loggers, etc. A TargetResult is not defined in Framework (but in MS.Build.dll) so it's not as suitable as a "currency" that everyone understands.

Reply:
All good points :)